[BUG] FE start failed when replaying InsertOverwriteJob state change log

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. Build a cluster of version 2.4.1.
2. create many mv tables and load data to the table to trigger mv refresh.
3. Restart FE leader during the load process.
4. upgrade to 2.4.3.

### Expected behavior (Required)

### Real behavior (Required)
```
2023-02-21 23:20:10,217 WARN (replayer|69) [GlobalStateMgr.replayJournalInner():1723] catch exception when replaying 37314317,
com.starrocks.journal.JournalInconsistentException: failed to load journal type 10096
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:954) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1712) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1571) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1636) [starrocks-fe.jar:?]
Caused by: java.lang.NullPointerException
        at com.starrocks.load.InsertOverwriteJobRunner.<init>(InsertOverwriteJobRunner.java:67) ~[starrocks-fe.jar:?]
        at com.starrocks.load.InsertOverwriteJobManager.replayInsertOverwriteStateChange(InsertOverwriteJobManager.java:142) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:927) ~[starrocks-fe.jar:?]
        ... 4 more
2023-02-21 23:20:10,232 WARN (replayer|69) [GlobalStateMgr$5.runOneCycle():1574] got interrupt exception or inconsistent exception when replay journal 37314317, will exit,
com.starrocks.journal.JournalInconsistentException: failed to load journal type 10096
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:954) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1712) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1571) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1636) [starrocks-fe.jar:?]
Caused by: java.lang.NullPointerException
        at com.starrocks.load.InsertOverwriteJobRunner.<init>(InsertOverwriteJobRunner.java:67) ~[starrocks-fe.jar:?]
        at com.starrocks.load.InsertOverwriteJobManager.replayInsertOverwriteStateChange(InsertOverwriteJobManager.java:142) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:927) ~[starrocks-fe.jar:?]
        ... 4 more
```

### StarRocks version (Required)
 - 2.4.3 2.4.4

--- a/fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobManager.java
+++ b/fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobManager.java
@@ -152,6 +152,10 @@ public class InsertOverwriteJobManager implements Writable, GsonPostProcessable 
 
     public void replayInsertOverwriteStateChange(InsertOverwriteStateChangeInfo info) {
         InsertOverwriteJob job = getInsertOverwriteJob(info.getJobId());
+        if (job == null) {
+            LOG.info("cannot find job: {}, ignore", info);
+            return;
+        }
         InsertOverwriteJobRunner runner = new InsertOverwriteJobRunner(job);
         runner.replayStateChange(info);
         if (job.isFinished()) {

